libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/NavigationViewEventDispatcher.java

```diff
@@ -155,14 +166,30 @@ void onInstructionListVisibilityChanged(boolean shown) {
   }
 
   private void assignProgressChangeListener(NavigationViewOptions navigationViewOptions, MapboxNavigation navigation) {
-    if (navigationViewOptions.progressChangeListener() != null) {
-      navigation.addProgressChangeListener(navigationViewOptions.progressChangeListener());
+    ProgressChangeListener progressChangeListener = navigationViewOptions.progressChangeListener();
+    if (progressChangeListener != null) {
+      this.progressChangeListener = progressChangeListener;
```

Comment:
We can move this assignment to L#69 like
```java
this.progressChangeListener = navigationViewOptions.progressChangeListener();
```
and remove this line. It's safe because then we're checking if the listener is `null` in `removeProgressChangeListener` 👇